test/org/javarosa/xpath/expr/XPathPathExprCurrentTest.java

```diff
@@ -15,86 +15,49 @@
 
-import org.javarosa.core.model.FormDef;
-import org.javarosa.core.model.ItemsetBinding;
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.Matchers.is;
+import static org.javarosa.xpath.expr.AnswerDataMatchers.answer;
+import static org.javarosa.xpath.expr.SelectChoiceMatchers.choice;
+import static org.junit.Assert.assertThat;
+
+import java.util.List;
 import org.javarosa.core.model.SelectChoice;
-import org.javarosa.core.model.data.SelectOneData;
-import org.javarosa.core.model.data.StringData;
-import org.javarosa.core.model.data.helper.Selection;
-import org.javarosa.core.model.instance.InstanceInitializationFactory;
-import org.javarosa.core.test.FormParseInit;
-import org.javarosa.form.api.FormEntryController;
-import org.javarosa.form.api.FormEntryModel;
-import org.javarosa.form.api.FormEntryPrompt;
 import org.junit.Before;
 import org.junit.Test;
 
-import static junit.framework.TestCase.assertEquals;
-import static org.javarosa.test.utils.ResourcePathHelper.r;
-
 public class XPathPathExprCurrentTest {
-    private FormDef formDef;
-    private FormEntryController formEntryController;
+    private Scenario scenario;
 
     @Before
     public void setUp() {
-        FormParseInit fpi = new FormParseInit(r("relative-current-ref.xml"));
-        formDef = fpi.getFormDef();
-        formDef.initialize(true, new InstanceInitializationFactory());
-        FormEntryModel formEntryModel = new FormEntryModel(formDef);
-        formEntryController = new FormEntryController(formEntryModel);
+        scenario = Scenario.init("relative-current-ref.xml");
     }
 
-    /**
-     * current() in a calculate should refer to the node it is in (in this case, /data/my_group/name_relative).
-     * This means that to refer to a sibling node, the path should be current()/../<name of sibling node>. This is
-     * verified by changing the value of the node that the calculate is supposed to refer to
-     * (/data/my_group/name) and seeing that the dependent calculate is updated accordingly.
-     */
     @Test
     public void current_in_calculate_should_refer_to_node() {
-        formEntryController.jumpToFirstQuestionWithName("name");
-        StringData nameValue = new StringData("Bob");
-        formEntryController.answerQuestion(nameValue, true);
-
-        StringData relativeName = (StringData) formDef.getFirstDescendantWithName("name").getValue();
-        assertEquals(nameValue.getValue(), relativeName.getValue());
+        scenario.answer("/data/my_group/name", "Bob");
+
+        // The binding of /data/my_group/name_relative is:
+        //   <bind calculate="current()/../name" nodeset="/data/my_group/name_relative" type="string"/>
+        // That will copy the value of our previous answer to /data/my_group/name
+        assertThat(
+            scenario.answerOf("/data/my_group/name_relative"),
+            is(answer(scenario.answerOf("/data/my_group/name")))
+        );
     }
 
-    /**
```

Comment:
@ggalmazor Can you say more about why you decided to remove these comments? I don't think the reasoning for why these tests exist is so clear without them.

Reply:
I was trying to replace the abstract description of how that works with a good enough concrete example & assertion combination. There's also a comment explaining how the itemset is defined in the form, above the assertion.

Since you've asked, I feel like this is not enough :) If you want, I can bring back those comments.

Anyway, after reviewing the test, I think the method name is not very good... It should  say `current_as_itemset_choice_filter_root_should_refer_to_the_select_node()`. The previous test method in the same test class could be renamed to `current_as_calculate_root_should_refer_to_its_bound_nodeset()` as well.

I think you were great at explaining the behavior of `current`. Maybe we could also add them where the `current` is actually processed at `XPathPathExpr:129`